src/renderer/components/snipping-tool.tsx

```diff
@@ -0,0 +1,167 @@
+import { ipcRenderer } from 'electron';
+import * as React from 'react';
+import { i18n } from '../../common/i18n-preload';
+
+interface IState {
+  snipImage: string;
+  height: number;
+  width: number;
+}
+
+const SNIPPING_TOOL_NAMESPACE = 'ScreenSnippet';
+
+export default class SnippingTool extends React.Component<{}, IState> {
+  private readonly eventHandlers = {
+    onDraw: () => this.draw(),
+    onHighlight: () => this.highlight(),
+    onErase: () => this.erase(),
+    onDone: () => this.done(),
+    onUndo: () => this.undo(),
+    onRedo: () => this.redo(),
+    onClear: () => this.clear(),
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      snipImage: 'Screen-Snippet',
+      height: 600,
+      width: 800,
+    };
+    this.updateState = this.updateState.bind(this);
+  }
+
+  /**
+   * Renders the Snipping tool
+   */
+  public render(): JSX.Element {
+    const { snipImage } = this.state;
+    return (
+      <div className='SnippingTool' lang={i18n.getLocale()}>
+        <header>
+          <div className='DrawActions'>
+            <button
+              className='ActionButton'
+              onClick={this.eventHandlers.onDraw}
+            >
+              <img src='../renderer/assets/snip-draw.svg' />
+            </button>
+            <button
+              className='ActionButton'
+              onClick={this.eventHandlers.onHighlight}
+            >
+              <img src='../renderer/assets/snip-highlight.svg' />
+            </button>
+            <button
+              className='ActionButton'
+              onClick={this.eventHandlers.onErase}
+            >
+              <img src='../renderer/assets/snip-erase.svg' />
+            </button>
+          </div>
+          <div className='ClearActions'>
+            <button
+              className='ActionButton'
+              onClick={this.eventHandlers.onUndo}
+            >
+              <img src='../renderer/assets/snip-undo.svg' />
+            </button>
+            <button
+              className='ActionButton'
+              onClick={this.eventHandlers.onRedo}
+            >
+              <img src='../renderer/assets/snip-redo.svg' />
+            </button>
+            <button
+              className='ClearButton'
+              onClick={this.eventHandlers.onClear}
+            >
+              Clear
+            </button>
+          </div>
+        </header>
+
+        <main>
```

Comment:
When presenting screen shots in the snippet tool, the screen shot is of different dimensions than what is being screen shoted. When the screen shot gets sent, it is in the original dimensions. This is actually a blocker for adding drawing capabilities, since the x and y coordinates are not the same in the tool and in the real screen shot, meaning when we add a SVG shape on and X and Y coordinate in the tool, those X and Y coordinates marks different things in the tool and in the real screen shot.


The screen shot presented in the tool should be of 1:1 dimensions as of what is being screen shoted in all cases, except if the screen shot is too big to fit the tool. In that case, the screen shot should be shrunk down (but of course keep the ratio). If the screen shot is being shown in different dimensions than its real dimensions, than the shapes will be off. Therefore, if the screen shot does not fit the tool, it must be shrunk down, not just change the presented dimensions, because of what I wrote before.


I think the wrong dimensions is only applied if the screen shot height > width. This can also be seen because  if heigth > width the screen shot gets a bit pixelated (tell tale sign that it is not beign shown in its correct dimensions). If width > heigth it is not.

Reply:
Now img size is the same in tool and what is exported, but it is shrunk down in regards to what is actually being screen shoted. Creating a bug for this to track it.